test/integration/discovery-consul.bats

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bats
+
+load helpers
+
+# Address on which Consul will listen (random port between 8000 and 9000).
+CONSUL_HOST=127.0.0.1:$(( ( RANDOM % 1000 )  + 8000 ))
+
+# Container name for integration test
+CONTAINER_NAME=swarm_consul
+
+function check_leader() {
+	# Confirm Cluster leader election
+	docker_host logs $CONTAINER_NAME | grep "New leader elected"
+	# Check member state
+	docker_host logs $CONTAINER_NAME | grep "consul: member '$CONTAINER_NAME' joined, marking health alive"
+}
+
+function start_consul() {
+	run docker_host run --name=$CONTAINER_NAME -h $CONTAINER_NAME -p $CONSUL_HOST:8500 -d progrium/consul -server -bootstrap-expect 1 -data-dir /test
```

Comment:
Small nit: There is no need to use `run` here and checking $status -eq 0 if you are expecting this to work.

Without `run`, any command that exits with non-zero will fail the test